core/common/assets/js/api/core/data/cache.js

```diff
@@ -0,0 +1,210 @@
+import LocalStorage from './stroages/local-storage';
+
+export default class Cache {
+	/**
+	 * Function constructor().
+	 *
+	 * Create cache.
+	 *
+	 * @param {Data} manager
+	 */
+	constructor( manager ) {
+		this.manager = manager;
+
+		this.storage = new LocalStorage();
+	}
+
+	validateRequestData( requestData ) {
+		if ( 'object' !== typeof requestData.component ) {
+			throw new Error( 'Invalid requestData.component.' );
+		}
+
+		if ( 'string' !== typeof requestData.command ) {
+			throw new Error( 'Invalid requestData.command.' );
+		}
+
+		if ( 'string' !== typeof requestData.endpoint ) {
+			throw new Error( 'Invalid requestData.endpoint.' );
+		}
+	}
+
+	/**
+	 * Function receive().
+	 *
+	 * Receive from cache.
+	 *
+	 * @param {RequestData} requestData
+	 *
+	 * @return {(Promise|boolean)}
+	 */
+	receive( requestData ) {
+		this.validateRequestData( requestData );
+
+		const data = this.get( requestData );
+
+		if ( null !== data ) {
+			// If data comes from cache, add 'receiveCache' to requestData.
+			requestData.receiveCache = true;
+
+			return new Promise( async ( resolve ) => {
+				resolve( data );
+			} );
+		}
+
+		// TODO: Check if possible, always return promise and reject it.
+		return false;
+	}
+
+	/**
+	 * Function load().
+	 *
+	 * Load data to cache
+	 *
+	 * @param {RequestData} requestData
+	 * @param {*} data
+	 */
+	load( requestData, data ) {
+		this.validateRequestData( requestData );
+
+		const componentName = requestData.component.getNamespace(),
+			nakedEndpoint = requestData.endpoint.replace( componentName + '/', '' ),
+			nakedEndpointParts = nakedEndpoint.split( '/' );
+
+		let newData = {};
+
+		// Analyze reaming endpoint.
+		if ( nakedEndpointParts.length && nakedEndpoint !== componentName ) {
+			// Using reaming endpoint build new data object.
+			const result = nakedEndpointParts.reduce( ( accumulator, nakedEndpointPart ) => {
+				accumulator[ nakedEndpointPart ] = {};
+
+				return accumulator[ nakedEndpointPart ];
+			}, newData );
+
+			// 'result' is equal to 'newData' with a deeper pointer, build based on 'nakedEndpointParts' ( will effect newData ).
+			Object.assign( result, data );
+		} else {
+			newData = data;
+		}
+
+		let oldData = this.storage.getItem( componentName );
+
+		if ( oldData !== null ) {
+			oldData = JSON.parse( oldData );
+
+			newData = jQuery.extend( true, oldData, newData );
```

Comment:
`extend` doesn't removes outdated data. for example: 
`oldData = { a: 1 }, newData = { b: 2 };`
After extending the new data is `{ a: 1, b: 2 }`, but `a` should be deleted.